app/helpers/check_answers_helper.rb

```diff
@@ -1,39 +1,67 @@
 module CheckAnswersHelper
-  def get_answered_questions_total(subsection_pages, case_log)
-    questions = subsection_pages.values.flat_map do |page|
-      page["questions"].keys
+  def total_answered_questions(subsection, case_log)
+    total_questions(subsection, case_log).keys.count do |question_key|
+      case_log[question_key].present?
     end
-    questions.count { |question| case_log[question].present? }
   end
 
-  def get_total_number_of_questions(subsection_pages)
-    questions = subsection_pages.values.flat_map do |page|
-      page["questions"].keys
+  def total_number_of_questions(subsection, case_log)
+    total_questions(subsection, case_log).count
+  end
+
+  def total_questions(subsection, case_log)
+    form = Form.new(2021, 2022)
+    questions = form.questions_for_subsection(subsection)
+    questions_not_applicable = []
+    questions.reject do |question_key, question|
+      question.fetch("conditional_for", []).map do |conditional_question_key, condition|
+        if condition_not_met(case_log, question_key, question, condition)
+          questions_not_applicable << conditional_question_key
+        end
+      end
+      questions_not_applicable.include?(question_key)
+    end
+  end
+
+  def condition_not_met(case_log, question_key, question, condition)
```

Comment:
It's not ideal that we're now evaluating these conditions in two places (Ruby side and JS side), and that the evaluation of them is relatively brittle, but given how few use cases we currently have I think it's OK to leave refactoring this out until we have more and/or a better abstraction becomes clear.